ui/src/pages/origination/NewBinaryOption.tsx

```diff
@@ -55,29 +55,27 @@ const NewBinaryOptionComponent = ({ history }: RouteComponentProps) => {
       new Date(d.getTime() - d.getTimezoneOffset() * 60000).toISOString().slice(0, 10)) ||
     '';
 
-  const obsToday: Observation<DamlDate, Id> = { tag: 'DateIdentity', value: {} };
-  const obsExpiry: Observation<DamlDate, Id> = { tag: 'DateConst', value: parseDate(expiry) };
-  const obsEuropean: Observation<DamlDate, Id> = {
-    tag: 'DateEqu',
-    value: { _1: obsToday, _2: obsExpiry },
+  const ineqEuropean: Inequality<DamlDate, Id> = {
+    tag: 'TimeGte',
+    value: parseDate(expiry),
   };
-  const obsStrike: Observation<DamlDate, Id> = { tag: 'DecimalConst', value: strike };
-  const obsSpot: Observation<DamlDate, Id> = { tag: 'DecimalObs', value: underlying };
-  const obsPayoff: Observation<DamlDate, Id> = {
-    tag: 'DecimalLte',
+  const obsStrike: Observation<DamlDate, string> = { tag: 'Const', value: { value: strike } };
```

Comment:
The type here and below is actually `DamlDate, Decimal`.

Is it ok to leave it as `string`, or is there a better way to type this?

Reply:
`DamlDate` is a `string` in Typescript :

`import { Date as DamlDate } from '@daml/types';`

From the imported library "@daml/types" -> `export declare type Date = string;`

The Daml type `Decimal` is also a `string`, we can see in the "@daml/types" library that this is set as :

`export declare type Numeric = string;`
`export declare type Decimal = Numeric;`

The explanation in the library for typing `Decimal`s as `string`s in Typescript is :

> The counterpart of Daml's `Numeric` type. We represent `Numeric`s as string in order to avoid a loss of precision. The string must match the regular expression `-?(?:0|[1-9]\d*)(?:\.\d+)?(?:[eE][+-]?\d+)?`.

**TR;DR** - you're safe as the daml library is typing both of those Daml types as `string`s in Typescript although you could set this as `Observation<DamlDate, Decimal>` and import `Decimal` from the daml typescript library to be more explicit.